Rudimentary worktree support

Robinhola opened the pull request:
reopening https://github.com/palantir/gradle-git-version/pull/261/files

palantirtech commented:
Thanks for your interest in [palantir/gradle-git-version](https://github.com/palantir/gradle-git-version), @Robinhola! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

Commit:
Git-worktree elimentary support

Credits to @aioobe and in his
https://github.com/palantir/gradle-git-version/pull/261 PR.

The idea here is just to fallback to native git in case we find
ourselves using a git worktree.

Commit:
Add changelog

Commit:
WIP add test - issue: jgit does not initialise on a worktree

Commit:
wip hack
